SampleViewer/DownloadsManager.h

```diff
@@ -0,0 +1,154 @@
+
+public:
+  explicit DownloadsManager(QObject* parent = nullptr);
+  ~DownloadsManager() override;
+
+  // Initialization
+  void initialize(Esri::ArcGISRuntime::Portal* portal, const QString& homePath);
+
+  // Download operations
+  void downloadAllDataItems();
+  void downloadDataItemsForSample(Sample* sample);
+  void cancelAllDownloads();
+
+  // Delete operations
+  bool deleteAllOfflineData();
+  bool deleteProjectOfflineData(const QString& sampleName);
+
+  // Query methods
+  bool hasOfflineData(const QString& sampleName);
+  OfflineDataProjectsModel* getOfflineDataProjects();
+  bool hasAnyDataToDownload();
+  bool hasAnyDataToDelete();
+
+  // State getters
+  bool downloadInProgress() const;
+  bool isBulkDownload() const;
+  QString downloadText() const;
+  double downloadProgress() const;
+  bool cancelDownload() const;
+  bool downloadFailed() const;
+  OfflineDataProjectsModel* offlineDataProjects() const;
+
+  // State setters
+  void setCancelDownload(bool cancel);
+
+  void setSamples(SampleListModel* samples)
+  {
+    m_samples = samples;
+  }
+
+signals:
+  void downloadInProgressChanged();
+  void isBulkDownloadChanged();
+  void downloadTextChanged();
+  void downloadProgressChanged();
+  void cancelDownloadChanged();
+  void downloadFailedChanged();
+  void offlineDataStateChanged();
+  void doneDownloadingChanged();
+
+private:
+    static constexpr int PROGRESS_UPDATE_INTERVAL_MS = 200;
+
+    struct SampleDownloadState
+    {
+        int totalItems = 0;
+        int downloadedItems = 0;
+        bool downloaded = false;
+        bool isDownloading = false;
+        double progress = 0.0;
+    };
+  SampleDownloadState getSampleDownloadState(Sample* sample) const;
+
+  int countDownloadedItems(Sample *sample) const;
+  bool isSampleDownloading(Sample* sample) const;
+  void deleteDataItemFile(DataItem* dataItem);
+  void cancelDataItem(const QString& dataItemKey);
+  void cleanupEmptyParentDirectories(const QString& startPath, const QString& boundary);
+
+  QJsonObject loadDownloadTracking();
+  void saveDownloadTracking(const QJsonObject& data);
+  void trackFilesForItem(const QString& itemId, const QStringList& files);
+
+  void downloadNextDataItem();
+  void fetchPortalItemData(const QString& itemId, const QString& outputPath);
+  void updateOfflineDataProjects();
+  double calculateSampleDownloadProgress(Sample* sample) const;
+  QString formattedPath(const QString& outputPath, const QString& folderName = QString());
+
+  void setDownloadInProgress(bool inProgress);
+  void setIsBulkDownload(bool isBulk);
+  void setDownloadText(const QString& text);
+  void setDownloadProgress(double progress);
+  void setDownloadFailed(bool failed);
+
+  QQueue<DataItem *> m_dataItems;
+  QMap<QString, double> m_dataItemProgress;
+  QMap<QString, Esri::ArcGISRuntime::PortalItem *> m_activeDownloads;
+  QTimer *m_progressUpdateTimer = nullptr;
```

Comment:
Style consistency please. There are many other lines here that should be standardized as well.

```suggestion
  QTimer* m_progressUpdateTimer = nullptr;
```